src/main/java/com/exadel/discountwebapp/location/entity/Country.java

```diff
@@ -0,0 +1,31 @@
+package com.exadel.discountwebapp.location.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import org.apache.commons.lang3.builder.ToStringExclude;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.*;
+import java.util.List;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Table(name = "country")
+@EntityListeners(AuditingEntityListener.class)
+public class Country {
+    @EqualsAndHashCode.Exclude
+    @Id
+    @Column(name = "country_code", length = 50, nullable = false, unique = true)
+    String countryCode;
+
+    @Column(name = "country_full_name", length = 255, nullable = false, unique = true)
+    String countryFullName;
+
+    @ToStringExclude
+    @OneToMany(mappedBy = "countryCode")
+    private List<Location> locations;
```

Comment:
Lookup table should have only dictionary information. It should not has any of business logic. 

It is better to remove this field and make location -> country relationship unidirectional

Reply:
Removed